### PR TITLE
 OP-103: add support for ResultSet.getString() method to retrieve different SQL types.

### PR DESCRIPTION
**Pull request by JadenW**

### CODE REVIEW

File: src/frontend/org/voltdb/jdbc/JDBC4ResultSet.java [hunk 1]
```diff
@@ -790,7 +790,14 @@ public Statement getStatement() throws SQLException {
     public String getString(int columnIndex) throws SQLException {
         checkColumnBounds(columnIndex);
         try {
-            return table.getString(columnIndex - 1);
+            VoltType type = table.getColumnType(columnIndex - 1);
```
Comment (akhanzode) [inline src/frontend/org/voltdb/jdbc/JDBC4ResultSet.java]: may be wrap this into getObject().toString()

Comment (JadenW) [inline src/frontend/org/voltdb/jdbc/JDBC4ResultSet.java]: There is a need for special treatment for type VarBinary,
getObject methods will return Byte[], toString() method wont return actual string but only HashValue.
So right now I decode that by creating new String Object and assign byte[] to it.  
Also , is "UTF-8” decoding need here?

Comment (akhanzode) [inline src/frontend/org/voltdb/jdbc/JDBC4ResultSet.java]: What does other database drivers do? We should be consistent. Also it looks like ou found issue with getObject can you fix that in consistent manner?

Comment (JadenW) [inline src/frontend/org/voltdb/jdbc/JDBC4ResultSet.java]: Mysql could specify an "encoding" for each byte[] filed.  
However votldb always use hex-encoded binary value to represent String input. So I think just to use Encoder.hexEncode() from volt.utils package for covert byte[] to String will be fine.  (This is method is also used for putJSONRep() method, which is used for generating string from bytes[] to JSON string)

I think this is just byte[] to String conversion issue, so getObject method should be fine.

**Comment by akhanzode** [comment]
squash merged.

**Pull request closed without merging**